CeeFM/Decideify
# Music.js Algorithm Implementation and Testing Part 3

**Description**
Music is the final content hub that is now up and running with a V1 of the suggestion algorithm! You can submit a genre or request a random suggestion, and a single album is returned for your review and consideration. You can cycle through the other returned suggestions for your query, save any of the individual recommendations to your database/account, or you can start over with a new query entirely.

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- On the music.js page:
  - You can choose a genre / type for the piece of content from a dropdown menu on the music hub page, or select random
  - Clicking the 'Decideify Music For Me" should bring up an image of a golden record player encrusted with diamonds and an equally ritzy record on top of it that says YOUR NEW FAVORITE ALBUM as well as a button below the image that says Show Me My Music Suggestion!
  - Clicking the Show Me My Music Suggestion! button will return the details about an individual album from the API results. You will also see new buttons for Show Me Another Music Suggestion! or Save Album.
  - Show Me Another Music Suggestion! will cycle through the other ~100 albums on the page, with a possibility of repeats - a future pull request will hopefully address this is and prevent repeat suggestions - and simply not allow any further cycling once all suggestions have been reviewed.
  - The Save Suggestion button should save the album to your Decideify account/database

Checklist
